File: leetcode/452. Minimum Number of Arrows to Burst Balloons.cpp
class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {
        if(points.size() == 0) {
            return 0;
        }
        
        sort(points.begin(), points.end(), [](vector<int>& p1, vector<int>& p2){
            return p1[0] < p2[0];
        });
        
        int cnt = 0;
        auto curr = points[0];
        for(size_t i = 1; i < points.size(); ++i){
            if(curr[1] < points[i][0]){
                cnt += 1;
                curr[0] = points[i][0];
                curr[1] = points[i][1];
            } else {
                curr[0] = points[i][0];
                curr[1] = min(curr[1], points[i][1]);
            }
        }
        
        return cnt + 1;
    }
};

File: leetcode/452. Minimum Number of Arrows to Burst Balloons.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        if len(points) == 0:
            return 0
        
        points.sort(key=lambda point: point[0])
        curr = points[0]
        i = 1
        cnt = 0
        while i < len(points):
            if points[i][0] > curr[1]:
                cnt += 1
                curr = points[i]
            else:
                curr = [points[i][0], min(curr[1], points[i][1])]
            i += 1
        
        return cnt + 1
